core/tests/ts-integration/tests/system.test.ts

```diff
@@ -337,6 +361,78 @@ describe('System behavior checks', () => {
         }
     });
 
+    test('Should check that createEVM is callable and executes successfully', async () => {
+        if (!process.env.RUN_EVM_TEST) {
+            console.log('skip createEVM test');
+            return;
+        }
+
+        if (!isETHBasedChain) {
+            const baseTokenDetails = testMaster.environment().baseToken;
+            const baseTokenMaxAmount = await alice.getBalanceL1(baseTokenDetails.l1Address);
+            await (await alice.approveERC20(baseTokenDetails.l1Address, baseTokenMaxAmount)).wait();
+        }
+
+        const contractInitcode: BytesLike = ethers.hexlify('0x69602a60005260206000f3600052600a6016f3');
+
+        const abi = ['function createEVM(bytes _initCode)'];
+
+        const iface = new ethers.Interface(abi);
+
+        const ContractDeployerCalldata: BytesLike = iface.encodeFunctionData('createEVM', [contractInitcode]);
+
+        const gasPrice = await scaledGasPrice(alice);
+        const l2GasLimit = maxL2GasLimitForPriorityTxs(testMaster.environment().priorityTxMaxGasLimit);
+
+        let priorityOpHandle = await alice.requestExecute({
+            contractAddress: await zksync.utils.CONTRACT_DEPLOYER_ADDRESS,
+            calldata: ContractDeployerCalldata,
+            l2GasLimit: l2GasLimit,
+            mintValue: isETHBasedChain ? 0n : expectedL2Costs,
+            overrides: {
+                gasPrice
+            }
+        });
+
+        await priorityOpHandle.waitL1Commit();
+
+        await expect(priorityOpHandle).toBeAccepted();
+    });
+
+    test('Should check that create2EVM is callable and executes successfully', async () => {
+        if (!process.env.RUN_EVM_TEST) {
+            console.log('skip create2EVM test');
+            return;
+        }
+
+        const contractInitcode: BytesLike = '0x69602a60005260206000f3600052600a6016f3';
+
+        const abi = ['function create2EVM(bytes32 _salt, bytes _initCode)'];
```

Comment:
could you unify the two abis into one and move to before the contracts? you can also probably create the interface there